scipy/special/_stirling2.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+
+from scipy.special._ufuncs import _stirling2
+from scipy.special._ufuncs import _stirling2_inexact
+
+
+def stirling2(n, k, exact=True):
+    if exact:
```

Comment:
Also, there are potential inputs where all choices of `n` with `1<k<n` will be overflow
so it seems like we'd want some sort of input validation method here to save the compute in `stirling2`.

Reply:
If `n` and `k` are restricted to 32 bit ints then ${n \brace n-1} = {n \choose 2}$ will never exceed the max 64 bit int, but that just means the condition check needs to be adjusted a little. This will be rendered moot by something you said in another comment.